x/proof/keeper/keeper.go

```diff
@@ -32,18 +43,45 @@ func NewKeeper(
 	authority string,
 
 	sessionKeeper types.SessionKeeper,
+	applicationKeeper types.ApplicationKeeper,
+	accountKeeper types.AccountKeeper,
 ) Keeper {
 	if _, err := sdk.AccAddressFromBech32(authority); err != nil {
 		panic(fmt.Sprintf("invalid authority address: %s", authority))
 	}
 
+	applicationQuerier := types.NewAppKeeperQueryClient(applicationKeeper)
+	accountQuerier := types.NewAccountKeeperQueryClient(accountKeeper)
+	// TODO_TECHDEBT: Use cosmos-sdk based polylog implementation once available. Also remove polyzero import.
+	polylogger := polylog.Ctx(context.TODO())
+
+	ringClientDeps := depinject.Supply(
+		polylogger,
+		applicationQuerier,
+		accountQuerier,
+	)
+
+	ringClient, err := rings.NewRingClient(ringClientDeps)
```

Comment:
Can we rename these to `ringKeeperClient` and the one below to `pubKeyKeeperClient`. When I was reading the business logic, I got worried that we were making networking requests from the `x/` repo which I believe happened due to the naming.

Reply:
I would be more in favour of removing the pubkeyKeeperClient - these are already accessible through the accountkeeper which can just be a dependency of the proof keeper itself.